File: Java-Kotlin/src/main/java/nowcoder/nc24.java
package nowcoder;

import common.datastructure.ListNode;

public class nc24 {
    public ListNode deleteDuplicates(ListNode head) {
        // write code here
        if (head == null || head.next == null) {
            return head;
        }

        ListNode dummy = new ListNode(-1);
        dummy.next = head;
        ListNode cur = head;
        ListNode prev = dummy;
        while (cur != null && cur.next != null) {
            if (cur.val == cur.next.val) {
                ListNode temp = cur.next;
                while (temp != null && temp.val == cur.val) {
                    temp = temp.next;
                }
                prev.next = temp;
                cur = temp;
            } else {
                prev = prev.next;
                cur = cur.next;
            }
        }
        return dummy.next;
    }
}

File: Go-Solutions/nowcoder/nc24.go
package nowcoder

func deleteDuplicates24(head *ListNode) *ListNode {
	// write code here
	if head == nil || head.Next == nil {
		return head
	}
	dummy := &ListNode{Val: -1, Next: head,}
	prev, cur := dummy, head
	for cur != nil && cur.Next != nil {
		if cur.Val == cur.Next.Val {
			temp := cur.Next
			for temp != nil && temp.Val == cur.Val {
				temp = temp.Next
			}
			prev.Next = temp
			cur = temp
		} else {
			prev = prev.Next
			cur = cur.Next
		}
	}
	return dummy.Next
}
